src/Opdex.Platform.WebApi/Controllers/IndexController.cs

```diff
@@ -99,21 +97,28 @@ public async Task<IActionResult> ResyncFromDeployment(ResyncFromDeploymentReques
 
     /// <summary>Rewind to Block</summary>
     /// <param name="request">Request to rewind back to specific block.</param>
+    /// <param name="cancellationToken">Cancellation token.</param>
     /// <response code="204">Indexer rewound.</response>
     [HttpPost("rewind")]
     [Authorize(Policy = "AdminOnly")]
     [ProducesResponseType(StatusCodes.Status204NoContent)]
     [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status403Forbidden)]
-    public async Task<ActionResult> Rewind(RewindRequest request)
+    public async Task<ActionResult> Rewind(RewindRequest request, CancellationToken cancellationToken)
     {
-        var tryLock = await _mediator.Send(new MakeIndexerLockCommand(IndexLockReason.Rewind), CancellationToken.None);
+        var latestSyncedBlock = await _mediator.Send(new RetrieveLatestBlockQuery(), cancellationToken);
+        if (latestSyncedBlock.Height <= request.Block) throw new InvalidDataException("block", "Block height must be less than current synced height.");
```

Comment:
This can be removed, we throw invalid data exception within `CreateRewindToBlock` command when we attempt this same logic. There are a couple locations that call that method, so I think the check fits within there rather than moving it out.